app/models/branch.rb

```diff
@@ -1,14 +1,35 @@
 # Model class Branch
 class Branch < ApplicationRecord
+  ### ASSOCIATIONS ###
+
+  has_one :address, as: :addressable, dependent: :destroy
+  accepts_nested_attributes_for :address, update_only: true, allow_destroy: true
+
   ### VALIDATIONS ###
 
   validates :name, :opening_time, :closing_time, presence: true
   validates :closing_time, comparison: { greater_than: :opening_time }, if: :closing_time
+  validates :address, presence: true
   validate :only_one_default, if: :default
 
+  ### CALLBACKS ###
+
+  before_update :validate_single_default_branch
+
   private def only_one_default
     if default_was != true && Branch.exists?(default: true)
       errors.add(:default, :present)
     end
   end
+
+  private def validate_single_default_branch
```

Comment:
Why have you created this callback?
we already have a validation for this.

Reply:
Yes fixed it